crates/nu-command/src/platform/ansi/ansi_.rs

```diff
@@ -673,129 +673,178 @@ Operating system commands:
-            for (k, v) in record {
-                match k.as_str() {
-                    "fg" => nu_style.fg = Some(v.as_string()?),
-                    "bg" => nu_style.bg = Some(v.as_string()?),
-                    "attr" => nu_style.attr = Some(v.as_string()?),
-                    _ => {
-                        return Err(ShellError::IncompatibleParametersSingle {
-                            msg: format!("unknown ANSI format key: expected one of ['fg', 'bg', 'attr'], found '{k}'"),
-                            span: code.span(),
-                        })
-                    }
+            match str_to_ansi(&code_string) {
+                Some(c) => c,
+                None => {
+                    return Err(ShellError::TypeMismatch {
+                        err_message: String::from("Unknown ansi code"),
+                        span: call
+                            .positional_nth(0)
+                            .expect("Unexpected missing argument")
+                            .span,
+                    })
                 }
             }
-            // Now create a nu_ansi_term::Style from the NuStyle
-            let style = nu_color_config::parse_nustyle(nu_style);
-            // Return the prefix string. The prefix is the Ansi String. The suffix would be 0m, reset/stop coloring.
-            style.prefix().to_string()
+        }
+    } else {
+        // This is a record that should look like
+        // { fg: "#ff0000" bg: "#00ff00" attr: bli }
+        let record = code.as_record()?;
+        // create a NuStyle to parse the information into
+        let mut nu_style = nu_color_config::NuStyle {
+            fg: None,
+            bg: None,
+            attr: None,
         };
-
-        Ok(Value::string(output, call.head).into_pipeline_data())
-    }
+        // Iterate and populate NuStyle with real values
+        for (k, v) in record {
+            match k.as_str() {
+                "fg" => nu_style.fg = Some(v.as_string()?),
+                "bg" => nu_style.bg = Some(v.as_string()?),
+                "attr" => nu_style.attr = Some(v.as_string()?),
+                _ => {
+                    return Err(ShellError::IncompatibleParametersSingle {
+                        msg: format!("unknown ANSI format key: expected one of ['fg', 'bg', 'attr'], found '{k}'"),
+                        span: code.span(),
+                    })
+                }
+            }
+        }
+        // Now create a nu_ansi_term::Style from the NuStyle
+        let style = nu_color_config::parse_nustyle(nu_style);
+        // Return the prefix string. The prefix is the Ansi String. The suffix would be 0m, reset/stop coloring.
+        style.prefix().to_string()
+    };
+    Ok(output)
 }
 
 pub fn str_to_ansi(s: &str) -> Option<String> {
     CODE_MAP.get(s).map(|x| String::from(*x))
 }
 
 fn generate_ansi_code_list(
-    engine_state: &EngineState,
+    engine_state: Option<&EngineState>,
```

Comment:
Instead of having two `Option<&EngineState/&StateWorkingSet>`  you could pass the ctrlc as a parameter directly. That would look cleaner.

Reply:
Good point. I'll do that.